plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -103,16 +103,30 @@ export async function createRouter<
         checks,
         entities,
       }: { checks: string[]; entities: CompoundEntityRef[] } = req.body;
+      const errors: string[] = [];
       const tasks = entities.map(async entity => {
         const entityTriplet =
           typeof entity === 'string' ? entity : stringifyEntityRef(entity);
-        const results = await factChecker.runChecks(entityTriplet, checks);
-        return {
-          entity: entityTriplet,
-          results,
-        };
+        try {
+          const results = await factChecker.runChecks(entityTriplet, checks);
+          return {
+            entity: entityTriplet,
+            results,
+          };
+        } catch (e: any) {
+          const errorMessage = `Failed to run check for entity ${entityTriplet} due to error: ${e.message}`;
```

Comment:
Again, I seriously think that it'll be very useful to use `https://github.com/backstage/backstage/blob/deb0e83ead40834130b45544043befea7f16c057/packages/errors/src/serialization/error.ts#L47` instead of a string in the first place. And placing it inside the entry as mentioned in the other comment.